models/samples/core/get_started/custom_tpuestimator.py

```diff
@@ -0,0 +1,197 @@
+                                 name="acc_op")
+  return {"accuracy": accuracy}
+
+
+def my_model(features, labels, mode, params):
+  """Deep Neural Network(DNN) model.
+
+  This is a DNN Model with 3 hidden layers. First 2 hidden layers are having
+  10 neurons in each. And number of neurons in the last layer is equal to the
+  number of output classes. This is a densely connected network where each
+  neuron of previous layer is connected to each neuron of next layer.
+
+  Args:
+    features: Feature values for input samples.
+    labels: label/class assigned to the corresponding input sample.
+    mode: "TRAIN"/"EVAL"/"PREDICT"
+    params: Dictionary used to pass extra parameters to model function from
+      the main function.
+
+  Returns:
+    TPUEstimator object.
+
+  """
+
+  # Create three fully connected layers.
+  net = tf.feature_column.input_layer(features, params["feature_columns"])
+  for units in params["hidden_units"]:
+    net = tf.layers.dense(net, units=units, activation=tf.nn.relu)
+
+  # Compute logits (1 per class).
+  logits = tf.layers.dense(net, params["n_classes"], activation=None)
+
+  # Compute predictions.
+  predicted_classes = tf.argmax(logits, 1)
+  if mode == tf.estimator.ModeKeys.PREDICT:
+    predictions = {
+        "class_ids": predicted_classes[:, tf.newaxis],
+        "probabilities": tf.nn.softmax(logits),
+        "logits": logits,
+    }
+    return tf.contrib.tpu.TPUEstimatorSpec(mode, predictions=predictions)
+
+  # Compute loss.
+  loss = tf.losses.sparse_softmax_cross_entropy(labels=labels,
+                                                logits=logits)
+
+  if mode == tf.estimator.ModeKeys.EVAL:
+    return tf.contrib.tpu.TPUEstimatorSpec(
+        mode=mode, loss=loss, eval_metrics=(metric_fn, [labels, logits]))
+
+  # Create training op.
+  if mode == tf.estimator.ModeKeys.TRAIN:
+    optimizer = tf.train.AdagradOptimizer(learning_rate=0.1)
+    if FLAGS.use_tpu:
+      optimizer = tf.contrib.tpu.CrossShardOptimizer(optimizer)
+    train_op = optimizer.minimize(loss, global_step=tf.train.get_global_step())
+    return tf.contrib.tpu.TPUEstimatorSpec(mode, loss=loss, train_op=train_op)
+
+
+def main(argv):
+  del argv  # Unused
+  if FLAGS.use_tpu:
+    assert FLAGS.model_dir.startswith("gs://"), ("'model_dir' should be a "
+                                                 "GCS bucket path!")
+
+  # Fetch the data
+  (train_x, train_y), (test_x, test_y) = iris_data.load_data()
+
+  # Feature columns describe how to use the input.
+  my_feature_columns = []
+  for key in train_x.keys():
+    my_feature_columns.append(tf.feature_column.numeric_column(key=key))
+
+  # Resolve TPU cluster and runconfig for this.
+  tpu_cluster_resolver = tf.contrib.cluster_resolver.TPUClusterResolver(
+      FLAGS.tpu)
+
+  run_config = tf.contrib.tpu.RunConfig(
+      model_dir=FLAGS.model_dir,
+      cluster=tpu_cluster_resolver,
+      session_config=tf.ConfigProto(
```

Comment:
You should be able to remove these options, e.g. session_config=tf.ConfigProto()

Reply:
Well. I prefer to keep as it will show that what config settings are required to move to GPU if TPU not available.